packages/core/rsc/shared-client/streamable.tsx

```diff
@@ -11,11 +12,28 @@ function assertStreamableValue(
     value.type !== STREAMABLE_VALUE_TYPE
   ) {
     throw new Error(
-      'Invalid value: this hook only accepts values created via `createStreamableValue` from the server.',
+      'Invalid value: this hook only accepts values created via `createStreamableValue`.',
     );
   }
 }
 
+function isStreamableValue(value: unknown): value is StreamableValue {
+  const hasSignature = !!(
+    value &&
+    typeof value === 'object' &&
+    'type' in value &&
+    value.type === STREAMABLE_VALUE_TYPE
+  );
+
+  if (!hasSignature && typeof value !== 'undefined') {
```

Comment:
Is `typeof value !== "undefined"` needed here? (since there is `typeof value === 'object'` above)

Reply:
Yeah it's intentional, I don't want to throw for the "lazy" case:

```js
const [streamableValue, setStreamableValue] = useState()
const [value] = useStreamableValue(streamableValue) // the value will be undefined initially

return <button onClick={async () => {
  const ret = await action()
  setStreamableValue(ret)
}}>
  ...
</button>
```